batchflow/models/eager_torch/base.py

```diff
@@ -0,0 +1,1203 @@
+                if isinstance(v, (torch.Tensor, torch.autograd.Variable)):
+                    v = v.detach().cpu().numpy()
+                output.append(v)
+            else:
+                raise KeyError('Unknown value to fetch', f)
+
+        output = output[0] if isinstance(fetches, str) else type(fetches)(output)
+        return output
+
+
+    def train(self, *args, feed_dict=None, fetches=None, use_lock=False, train_mode='',
+              accumulate_grads=True, sync_frequency=True, microbatch=None, **kwargs):
+        """ Train the model with the data provided
+
+        Parameters
+        ----------
+        args
+            Arguments to be passed directly into the model.
+        fetches : tuple, list
+            Sequence of tensor names to calculate and return.
+        use_lock : bool
+            If True, the whole train step is locked, thus allowing for multithreading.
+        train_mode : str or sequence of str
+            Name(s) of train step(s) to optimize. Regular expressions are allowed.
+            If multiple train steps are selected (either via passing a sequence or by using regular expression),
+            then all of them are optimized sequentially.
+        accumulate_grads : bool
+            If True, then gradients from different train modes are accumulated and applied once at the end.
+            If False, then gradients are applied for each of the train modes separately.
+        sync_frequency : int, bool or None
+            If int, then how often to apply accumulated gradients to the weights.
+            If True, then value from config is used (default value is to apply gradients after each batch of data).
+            If False or None, then gradients are applied after each batch of data.
+        microbatch : int, bool or None
+            If int, then size of chunks to split every batch into. Allows to process given data sequentially,
+            accumulating gradients from microbatches and applying them once in the end.
+            If True, then value from config is used (default value is not to use microbatching).
+            If False or None, then microbatching is not used.
+
+        Returns
+        -------
+        Calculated values of tensors in `fetches` in the same order.
+
+        Examples
+        --------
+        .. code-block:: python
+
+            model.train(B('images'), B('labels'), fetches='loss')
+        """
+        config = self.full_config
+        *inputs, targets = self._fill_input(*args, **{**(feed_dict or {}), **kwargs})
+
+        if sync_frequency is True:
+            sync_frequency = config['sync_frequency']
+        elif sync_frequency is False or sync_frequency is None:
+            sync_frequency = 1
+
+        if microbatch is not False:
+            if microbatch is True:
+                microbatch = config.get('microbatch', len(targets))
+            else:
+                microbatch = microbatch or config.get('microbatch', len(targets))
```

Comment:
`microbatch == None` comes here

Reply:
And changes to `config` option or no microbatching